boa/src/builtins/value/display.rs

```diff
@@ -102,6 +102,32 @@ pub(crate) fn log_string_from(x: &Value, print_internals: bool) -> String {
 
                     format!("[ {} ]", arr)
                 }
+                ObjectData::Map(ref map) => {
+                    let size = i32::from(
+                        &v.borrow()
+                            .properties()
+                            .get("size")
+                            .unwrap()
+                            .value
+                            .clone()
+                            .expect("Could not borrow value"),
+                    );
+                    if size == 0 {
+                        return String::from("Map(0)");
+                    }
+
+                    let mappings = map
+                        .borrow()
+                        .iter()
+                        .map(|(key, value)| {
+                            let key = log_string_from(key, print_internals);
+                            let value = log_string_from(value, print_internals);
+                            format!("{} → {}", key, value)
```

Comment:
To stop the overflow we need to check if the value is an `Map` object, if it is we print object `Map(sizeOfMap)` and we don't expand it any more, like SpiderMonkey does it:
```js
let x = new Map();
x.set("y", x);
x
```
This prints `Map { y → Map(1) }`

Reply:
I don't think this would work with all cases. For example
```javascript
let x = new Map();
let y = new Array();
y.push(x);
x.set("z", y);
x
```
will still overflow. Could we potentially fix this by only printing the top level items of collections, and have a boolean argument in log_string_from, like `log_string_from(x: &Value, print_internals: bool, top_level: bool) which is set to true when called outside of the method and set to false when called inside the method?

Reply:
> will still overflow. Could we potentially fix this by only printing the top level items of collections, and have a boolean argument in log_string_from, like `log_string_from(x: &Value, print_internals: bool, top_level: bool) which is set to true when called outside of the method and set to false when called inside the method?

Yes. would be a better solution for all cases printing something like:
```js
Map { y → Array(sizeOfArray) }
```

It is also nicer than having all the field of an array expanded. like:
```js
let x = new Map();
let y = new Array(1, 2, 3, 4, 5, 6, 7, 8);
x.set("z", y);
x.set("w", y);
x
```
this would be to long with the current implementation:
```js
Map { z → [1, 2, 3, 4, 5, 6, 7, 8], w  → [1, 2, 3, 4, 5, 6, 7, 8] }
```
instead it would be:
```js
Map { z → Array(8), w  → Array(8) }
```
This is mush cleaner :)

Reply:
Great! I'll have a go at implementing that.